### ISSUE TITLE
xmr-stak now in Gentoo; update REAMDE

### ISSUE THREAD

**Issue by candrews** [fireice-uk/xmr-stak#684]
xmr-stak is now in Gentoo so users can simply `emerge net-p2p/xmr-stak` to install it.

Can you please update the appropriate documentation so people interested in xmr-stak can know that?

**Comment by psychocrypt** [comment]
Thats great. Who is maintaining the sources and how often is the source
updated or is it mirrowed directly from github?

**Comment by psychocrypt** [comment]
i checked https://packages.gentoo.org/categories/net-p2p and can not find
xmr-stak

**Comment by candrews** [comment]
That site updates I think daily; it just hasn't noticed the new package.

It's in Gentoo, here's a link: https://github.com/gentoo/gentoo/tree/master/net-p2p/xmr-stak

I'm maintaining it (see `metadata.xml`).

Version 2.2.0 is in Gentoo as as well as -9999 version that pulls from the xmr-stak github repo. I'll monitor releases and add them to Gentoo as quickly as I can.

**Comment by psychocrypt** [comment]
thx that look really good. Please give me some time or open an pull request and add a gentoo compile section to our linux build documentation.

**Comment by fractalyse** [comment]
I tried it, and I've got and issue 

● xmr-stak.service - Unified all-in-one Monero miner
   Loaded: loaded (/usr/lib/systemd/system/xmr-stak.service; disabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Mon 2018-01-01 04:04:10 CET; 1s ago
  Process: 2629 ExecStart=/usr/bin/xmr-stak --cpu /etc/xmr-stak/cpu.config --amd /etc/xmr-stak/amd.config -c /etc/xmr-stak/main.config (code=exited, status=217/USER)
 Main PID: 2629 (code=exited, status=217/USER)

```
janv. 01 04:04:10 orus systemd[1]: Started Unified all-in-one Monero miner.
janv. 01 04:04:10 orus systemd[2629]: xmr-stak.service: Failed at step USER spawning /usr/bin/xmr-stak: Invalid argument
janv. 01 04:04:10 orus systemd[1]: xmr-stak.service: Main process exited, code=exited, status=217/USER
janv. 01 04:04:10 orus systemd[1]: xmr-stak.service: Unit entered failed state.
janv. 01 04:04:10 orus systemd[1]: xmr-stak.service: Failed with result 'exit-code'.
```

**Comment by candrews** [comment]
If you run `/usr/bin/xmr-stak --cpu /etc/xmr-stak/cpu.config --amd /etc/xmr-stak/amd.config -c /etc/xmr-stak/main.config` does it work? If not, what happens?

**Comment by fractalyse** [comment]
yes it works great, only systemd service isn't working with option PrivateUserS=true

**Comment by candrews** [comment]
What version of systemd are you using? Do you have /etc/nsswitch.conf set up correctly?

And a happy new year to you as well! 🎉

**Comment by fractalyse** [comment]
I'm using 233-r6 
```
# /etc/nsswitch.conf:
# $Header: /var/cvsroot/gentoo/src/patchsets/glibc/extra/etc/nsswitch.conf,v 1.2 2017/08/12 16:21:44 slyfox Exp $

passwd:      compat files
shadow:      compat files
group:       compat files

hosts:       files dns
networks:    files dns

services:    db files
protocols:   db files
rpc:         db files
ethers:      db files
netmasks:    files
netgroup:    files
bootparams:  files

automount:   files
aliases:     files
```

**Comment by candrews** [comment]
That's what I thought may be the problem. You need to make a bit of a modification, see https://www.freedesktop.org/software/systemd/man/nss-systemd.html

**Comment by fractalyse** [comment]
I've modified nsswitch.conf, and reboot, same issue.

**Comment by candrews** [comment]
Turns out DynamicUser was introduced in systemd 235... I only tested with the latest version; unfortunately I cannot help you with the version you're using (233).

**Comment by fractalyse** [comment]
Ok, thanks for your help

**Comment by nado** [comment]
It cant be built on a recent gentoo install (gcc-7), I dont have time right now to reinstall gcc-6 for one package sadly, are there any gcc-7 support plans ?

**Comment by candrews** [comment]
Can you please create a new issue with the failure information (logs) so we can figure out the problem? (If it is gcc 7 related, that's not Gentoo specific)

**Comment by candrews** [comment]
BTW, I'm not having any problems emerging xmr-stak-9999 on Gentoo and I'm using GCC 7 (`sys-devel/gcc-7.3.0-r1:7.3.0`).

**Comment by agentpatience** [comment]
Is there a document for gentoo install?

**Comment by candrews** [comment]
> Is there a document for gentoo install?

`emerge net-p2p/xmr-stak` should be all you need.

**Comment by agentpatience** [comment]
Thanks Mr. Andrews. I will give it ago. So far CentOS is my winning OS for hashrate but i'd like to try gentoo and arch to compare.

**Comment by agentpatience** [comment]
Sorry to be a broken record... I saw your original post. Is there an easy way to get a minimal version of gentoo installed with xmr-stak? I tried installing it manually and i spent all night trying to get everything running... i gave up after reading through the stage setup. DO you recommend the live image... or what is the best way to deploy gentoo quickly or is there such a thing... the minimal setup seems daunting.

**Comment by candrews** [comment]
I'm sorry, but this line of inquiry is off topic here.

Please refer to the appropriate [Gentoo support channels](https://www.gentoo.org/support/), there's a great forum, IRC channel, mailing list,and set of web documentation.

**Comment by agentpatience** [comment]
Hi Mr. Andrews. I once tried to get help on gentoo irc and they banned me for the simple questions I asked. I am hoping you can give me some personal insight. It doesn't have to be very lengthy and i'm sure someone else that sees this thread would like to know if there is a preferred base install method becausethe minimal install of gentoo put me through hell, should I even bother with the LiveDVD or is that not supported by USB install? I'm hoping that if I can get the desktop version up and running all I have to do is run emerge command?
